### PR TITLE
[WIP] read binlogs from solution and find latest msbuild 

### PR DESCRIPTION
**Pull request by colombod**
msbuild cannot be located unless using specific SKUS of VS 2017, now all vs fodler are scanned.

when loading a binlog from a build performed on a solution there are 0 results

### CODE REVIEW

File: tests/Buildalyzer.Tests/Integration/SolutionWithProjectsFixture.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                processRunner.Start();
+                processRunner.WaitForExit(60000);
+            }
+
+            using (ProcessRunner processRunner = new ProcessRunner("dotnet", $"build {solutionPath} /bl:{binlogPath}", Path.GetDirectoryName(solutionPath), environmentVariables, null))
+            {
+                processRunner.Start();
+                processRunner.WaitForExit(60000);
+            }
+            */
```
Comment (daveaglick) [inline tests/Buildalyzer.Tests/Integration/SolutionWithProjectsFixture.cs]: Any particular reason the binlog generation code here is commented out? Without it the tests obviously fail on a new checkout.

File: src/Buildalyzer/AnalyzerResult.cs [hunk 2]
```diff
@@ -108,16 +108,23 @@ internal AnalyzerResult(string projectFilePath, AnalyzerManager manager, Project
 
         internal void ProcessProject(ProjectStartedEventArgs e)
         {
-            // Add properties
-            foreach (DictionaryEntry entry in e.Properties.Cast<DictionaryEntry>())
+            if (e.Properties != null)
```
[the remark this answers is not in the document]
Comment (colombod) [reply src/Buildalyzer/AnalyzerResult.cs]: Yes I hit that with the tests I have added. I still need to work thisnproperly

File: src/Buildalyzer/Logging/EventProcessor.cs [hunk 3]
```diff
@@ -60,24 +67,43 @@ public EventProcessor(AnalyzerManager manager, ProjectAnalyzer analyzer, IEnumer
...
+            if (_projectFilePaths.Contains(normalizedProjectFilePath))
             {
                 // Get the TFM for this project
-                string tfm = e.Properties?.Cast<DictionaryEntry>()
-                    .FirstOrDefault(x => string.Equals(x.Key.ToString(), "TargetFrameworkMoniker", StringComparison.OrdinalIgnoreCase)).Value as string;
+                string tfm = GetTargetFramework(e);
                 if (!string.IsNullOrWhiteSpace(tfm))
                 {
-                    if (!_results.TryGetValue(tfm, out AnalyzerResult result))
+                    string key = $"{normalizedProjectFilePath}:{tfm}";
```
[the remark this answers is not in the document]
Comment (colombod) [reply src/Buildalyzer/Logging/EventProcessor.cs]: This exactly what i’d Like to design and discuss with you once back

File: tests/Buildalyzer.Tests/Integration/SolutionWithProjectsFixture.cs (hunk 1 above)
Comment (colombod) [reply tests/Buildalyzer.Tests/Integration/SolutionWithProjectsFixture.cs]: Is a work in progress and will execute that code. I just checked this is for you to see where I am trying to go with the changes

**Comment by daveaglick** [comment]
@colombod what are your thoughts on this PR? I'd like to do another release soon now that VS 2019 is out and I think it makes sense to include this solution work.

My thinking right now is that adding an extra `AnalyzerProjectResults` collection to gather results for each project is the way to go. I can do that plumbing.

Is there anything else you think needs to be done from your end before I start working the client API and merging this in?

File: src/Buildalyzer/Logging/EventProcessor.cs [hunk 4]
```diff
@@ -89,9 +115,29 @@ private void ProjectStarted(object sender, ProjectStartedEventArgs e)
             _currentResult.Push(null);
         }
 
+        private string GetTargetFramework(ProjectStartedEventArgs e)
+        {
+            string normalizedProjectFilePath = AnalyzerManager.NormalizePath(e.ProjectFile);
+            string tfm = null;
+            if (!_targets.TryGetValue(normalizedProjectFilePath, out tfm))
```
Comment (daveaglick) [inline src/Buildalyzer/Logging/EventProcessor.cs]: This part is confusing me. If I'm reading it correctly, we're caching the first TFM we see for each project file and always returning that as the TFM for that project file? That doesn't seem right... We want to always return the current project's TFM, right - that's why the `_results` key was changed to include TFM in the first place.

**Pull request closed without merging**